stubs/babel/babel/dates.pyi

```diff
@@ -1,30 +1,98 @@
-from datetime import date, datetime, time
-from typing import Any
+from datetime import date, datetime, time, timedelta, tzinfo
+from typing import Any, overload
+from typing_extensions import Literal
 
-NO_INHERITANCE_MARKER: str
-LC_TIME: Any
-date_ = date
-datetime_ = datetime
-time_ = time
+from babel.core import Locale
+from pytz import BaseTzInfo
+
+# The module contents here are organized the same way they are in the API documentation at
+# http://babel.pocoo.org/en/latest/api/dates.html
+
+# Date and Time Formatting
+def format_datetime(
+    datetime: datetime | None = ..., format: str = ..., tzinfo: tzinfo | None = ..., locale: str | Locale = ...
+) -> str: ...
+def format_date(date: date | datetime | None = ..., format: str = ..., locale: str | Locale = ...) -> str: ...
+def format_time(
+    time: time | datetime | None = ..., format: str = ..., tzinfo: tzinfo | None = ..., locale: str | Locale = ...
+) -> str: ...
+def format_timedelta(
+    delta: timedelta | int,
+    granularity: Literal["year", "month", "week", "day", "hour", "minute", "second"] = ...,
+    threshold: float = ...,
+    add_direction: bool = ...,
+    format: Literal["narrow", "short", "long"] = ...,
+    locale: str | Locale = ...,
+) -> str: ...
+def format_skeleton(
+    skeleton: str,
+    datetime: time | datetime | None = ...,
+    tzinfo: tzinfo | None = ...,
+    fuzzy: bool = ...,
+    locale: str | Locale = ...,
+) -> str: ...
+def format_interval(
+    start: datetime | date | time,
+    end: datetime | date | time,
+    skeleton: str | None = ...,
+    tzinfo: tzinfo | None = ...,
+    fuzzy: bool = ...,
+    locale: str | Locale = ...,
+) -> str: ...
 
-def get_timezone(zone: Any | None = ...): ...
-def get_next_timezone_transition(zone: Any | None = ..., dt: Any | None = ...): ...
+# Timezone Functionality
+
+@overload
+def get_timezone(zone: str | BaseTzInfo | None = ...) -> BaseTzInfo: ...
+@overload
+def get_timezone(zone: tzinfo) -> tzinfo: ...
+def get_timezone_gmt(
+    datetime: datetime | None = ...,
+    width: Literal["long", "short", "iso8601", "iso8601_short"] = ...,
+    locale: str | Locale = ...,
+    return_z: bool = ...,
+) -> str: ...
+def get_timezone_location(
+    dt_or_tzinfo: datetime | tzinfo | None = ..., locale: str | Locale = ..., return_city: bool = ...
```

Comment:
`dt_or_tzinfo` gets passed to `_get_dt_and_tzinfo()`, so it can also be a `str`, `int`, or `time`.